### PR TITLE
fix(spans): Normalize INP spans further

### PR DESCRIPTION
**Pull request by Zylphrex**
INP spans sets some top level span attributes inside span.data so make sure to pull them out to the top level. This has implications in the snuba consumer so that we do not redundant keys with slightly different names in span.data.

### CODE REVIEW

File: relay-server/src/services/processor/span/processing.rs [hunk 1]
```diff
@@ -590,6 +590,31 @@ fn normalize(
...
+            Some(Annotated(Some(Value::String(profile_id)), _)) => profile_id
+                .parse()
+                .map(|profile_id| EventId(profile_id))
+                .ok(),
+            _ => None,
+        } {
+            data.other.remove("profile_id");
+            span.profile_id = profile_id.into();
+        }
+    }
```
Comment (jjbayer) [inline relay-server/src/services/processor/span/processing.rs]: nit: Should we move this to a helper function (similar to `populate_ua_fields`)?

File: relay-server/src/services/processor/span/processing.rs [hunk 2]
```diff
@@ -590,6 +590,31 @@ fn normalize(
 
     populate_ua_fields(span, user_agent.as_deref(), client_hints.as_deref());
 
+    // INP spans sets some top level span attributes inside span.data so make sure to pull
+    // them out to the top level before further processing.
+    if let Annotated(Some(ref mut data), _) = span.data {
+        if let Some(exclusive_time) = match data.other.get("sentry.exclusive_time") {
```
Comment (jjbayer) [inline relay-server/src/services/processor/span/processing.rs]: Could you add `sentry.exclusive_time` and `profile_id` as fields on the `SpanData` type? I assume a hypothetical SDK that can only send Otel spans would still send these as span attributes? Or are these legacy fields that future SDKs should not send?

File: relay-server/src/services/processor/span/processing.rs [hunk 3]
```diff
@@ -1266,4 +1291,91 @@ mod tests {
...
+        normalize(&mut span, normalize_config()).unwrap();
+
+        let data = get_value!(span.data!);
+        assert!(!data.other.contains_key("sentry.exclusive_time"));
+        assert!(!data.other.contains_key("profile_id"));
+
+        assert_eq!(
+            get_value!(span.profile_id!),
+            &EventId("480ffcc911174ade9106b40ffbd822f5".parse().unwrap())
+        );
```
Comment (jjbayer) [inline relay-server/src/services/processor/span/processing.rs:1387]: Should this assert on `exclusive_time` as well?

File: relay-server/src/services/processor/span/processing.rs [hunk 4]
```diff
@@ -590,6 +590,31 @@ fn normalize(
 
     populate_ua_fields(span, user_agent.as_deref(), client_hints.as_deref());
 
+    // INP spans sets some top level span attributes inside span.data so make sure to pull
+    // them out to the top level before further processing.
+    if let Annotated(Some(ref mut data), _) = span.data {
```
Comment (jjbayer) [inline relay-server/src/services/processor/span/processing.rs]:
```suggestion
    if let Some(data) = span.data.value_mut() {
```

**Review by jjbayer — APPROVED** [review]

File: relay-server/src/services/processor/span/processing.rs (hunk 2 above)
Comment (Zylphrex) [reply relay-server/src/services/processor/span/processing.rs]: They can continue sending them as span attributes.

**Review by jjbayer — APPROVED** [review]

**Pull request merged**